### ISSUE TITLE
Support for an easy way to parallelize based on region

### ISSUE THREAD

**Issue by lonsbio** [ssadedin/bpipe#18]
_From [ssade...@gmail.com](https://code.google.com/u/106823827442770052555/) on 2012-04-04T11:16:49Z_

Currently Bpipe lets you easily parallelize if you split processing by sample or by splitting your input files up into pieces.   However many tools can operate on regions independently without splitting files up.   This is kind of tricky to do in Bpipe right now.   It would be nice to have some support for a direct syntax to say "run this pipeline with every chromosome in parallel".

Syntax:

  chr(1..22) \* [ call_variants ]

Will create a $chr variable that the call_variants stage can use.

More detailed regions could be created using an organism specific database:

  hg19.split(60) \* [ calculate_coverage_depth ]

This latter would figure out how to split the human genome into 60 roughly even parts for you and pass variables $chr, $start, $end to the calculate_coverage_depth pipeline stage, making it really easy to parallelize data processing.

_Original issue: http://code.google.com/p/bpipe/issues/detail?id=18_
